chainer/functions/normalization/batch_renormalization.py

```diff
@@ -0,0 +1,232 @@
+import warnings
+
+import numpy
+
+from chainer import configuration
+from chainer import cuda
+
+from chainer import function
+from chainer.utils import type_check
+
+
+def _as4darray(arr):
+    if arr.ndim == 0:
+        return arr.reshape(1, 1, 1, 1)
+    elif arr.ndim == 4:
+        return arr
+    else:
+        return arr.reshape(arr.shape[0], -1, 1, 1)
+
+
+def _xhat(x, mean, std, expander):
+    x_mu = x - mean[expander]
+    x_mu /= std[expander]
+    return x_mu
+
+
+class BatchRenormalizationFunction(function.Function):
+
+    def __init__(self, eps=2e-5, mean=None, var=None, decay=0.9,
+                 rmax=1, dmax=0, keep_r_d_fixed=False):
+        self.running_mean = mean
+        self.running_var = var
+        self.rmax = rmax
+        self.dmax = dmax
+        self.r = None
+        self.d = None
+        # Needed for gradient check to be possible
+        self.keep_r_d_fixed = keep_r_d_fixed
+
+        self.eps = eps
+        self.mean_cache = None
+        self.decay = decay
+
+    def check_type_forward(self, in_types):
+        n_in = type_check.eval(in_types.size())
+        if n_in != 3 and n_in != 5:
+            raise type_check.InvalidType(
+                '%s or %s' % (in_types.size() == 3, in_types.size() == 5),
+                '%s == %s' % (in_types.size(), n_in))
+        x_type, gamma_type, beta_type = in_types[:3]
+        M = type_check.eval(gamma_type.ndim)
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            x_type.ndim >= gamma_type.ndim + 1,
+            x_type.shape[1:1 + M] == gamma_type.shape,
+            # TODO(tkerola): Check shape
+            gamma_type.dtype == x_type.dtype,
+            beta_type.dtype == x_type.dtype,
+            gamma_type.shape == beta_type.shape,
+        )
+        if len(in_types) == 5:
+            mean_type, var_type = in_types[3:]
+            type_check.expect(
+                mean_type.dtype == x_type.dtype,
+                mean_type.shape == gamma_type.shape,
+                var_type.dtype == x_type.dtype,
+                var_type.shape == gamma_type.shape,
+            )
+
+    def forward(self, inputs):
+        xp = cuda.get_array_module(*inputs)
+        x, gamma, beta = inputs[:3]
+        if configuration.config.train:
+            if self.running_mean is None:
+                print("Set running_mean,var to zero")
```

Comment:
I do not think it is good idea that this function outputs this message because user could use stdout for other purpose and the message could contaminate the stdout.